src/Reflection/ReflectionProperty.php

```diff
@@ -118,6 +119,33 @@ private function getVisibilityAsString()
     }
 
     /**
+     * Set the default visibility of this propery. Use the core \ReflectionProperty::IS_* values as parameters, e.g.:
+     *
+     * @param int $newVisibility
+     * @throws \InvalidArgumentException
+     */
+    public function setVisibility($newVisibility)
+    {
+        $this->node->type = ~Class_::MODIFIER_PRIVATE;
```

Comment:
Hmm, I understood this to be:
 * remove the "private" bit, if set
 * remove the "protected" bit, if set
 * remove the "public" bit, if set
 * then in the switch statement, set the appropriate visibility bit depending on `$newVisibility`
Have I misunderstood?